dll/win32/kernel32/client/appcache.c

```diff
@@ -309,23 +309,78 @@ BasepCheckBadapp(IN HANDLE FileHandle,
 }
 
 /*
- * @unimplemented
+ * @implemented
  */
-VOID
+BOOL
 WINAPI
 BaseDumpAppcompatCache(VOID)
 {
-    STUB;
+    NTSTATUS Status;
+
+    Status = NtApphelpCacheControl(ApphelpCacheServiceDump, NULL);
+    return NT_SUCCESS(Status);
 }
 
+BOOLEAN
+WINAPI
+IsCallerAdminOrSystem(VOID)
+{
+    SID_IDENTIFIER_AUTHORITY SidAuthority = {SECURITY_NT_AUTHORITY};
+    PSID pSid = NULL;
+    NTSTATUS Status;
+    HMODULE hAdvapi32;
+    BOOL (WINAPI* pCheckTokenMembership)(HANDLE TokenHandle, PSID SidToCheck, PBOOL IsMember);
+    BOOL Allowed = FALSE;
+
+    hAdvapi32 = LoadLibraryW(L"advapi32.dll");
```

Comment:
This is insecure.
See for instance: https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/iphlpapi/iphlpapi_main.c;hb=d84ab9ec7c513eb0917bc9f24f6dc97b70e8485b#l2350